implementations/rust/ockam/ockam_abac/src/abac/abac.rs

```diff
@@ -31,79 +32,123 @@ pub const ABAC_IDENTIFIER_KEY: &str = "identifier";
 /// to verify if a policy expression is valid
 /// A similar access control policy is available as [`crate::policy::PolicyAccessControl`] where
 /// as [`crate::Policies`] can be used to retrieve a specific policy for a given resource and action
-pub struct AbacAccessControl {
+#[derive(Clone)]
+pub struct Abac {
     identities_attributes: Arc<IdentitiesAttributes>,
     authority: Identifier,
-    policy_expression: Expr,
     environment: Env,
 }
 
 /// Debug implementation printing out the policy expression only
-impl Debug for AbacAccessControl {
+impl Debug for Abac {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        let expression = &self.policy_expression;
-        f.write_str(format!("{expression:?}").as_str())
+        write!(f, "Authority: {}", self.authority)
     }
 }
 
-impl AbacAccessControl {
+impl Abac {
     /// Create a new AccessControl using a specific policy for checking attributes
     pub fn new(
         identities_attributes: Arc<IdentitiesAttributes>,
         authority: Identifier,
-        policy_expression: Expr,
         environment: Env,
     ) -> Self {
         Self {
             identities_attributes,
             authority,
-            policy_expression,
             environment,
         }
     }
+}
 
-    /// Create an AccessControl which will verify that the sender of
-    /// a message has an authenticated attribute with the correct name and value
-    pub fn create(
-        identities_attributes: Arc<IdentitiesAttributes>,
-        authority: Identifier,
-        attribute_name: &str,
-        attribute_value: &str,
-    ) -> AbacAccessControl {
-        let expression = List(vec![
-            Ident("=".into()),
-            Ident(format!("{SUBJECT_KEY}.{attribute_name}")),
-            Str(attribute_value.into()),
-        ]);
-        AbacAccessControl::new(identities_attributes, authority, expression, Env::new())
+impl Abac {
+    pub async fn get_outgoing_identifier(
+        ctx: &Context,
+        relay_msg: &RelayMessage,
+    ) -> Result<Option<Identifier>> {
+        let terminal = if let Some(terminal) = ctx
+            .find_terminal_address(relay_msg.onward_route().clone())
+            .await?
+        {
+            terminal
+        } else {
+            return Ok(None);
+        };
+
+        let identifier = if let Some(identifier) =
+            terminal
+                .metadata
+                .attributes
+                .iter()
+                .find_map(|(key, value)| {
+                    if key == IDENTITY_SECURE_CHANNEL_IDENTIFIER {
+                        Some(value.clone())
+                    } else {
+                        None
+                    }
+                }) {
+            identifier
+        } else {
+            return Ok(None);
+        };
+
+        let identifier = if let Ok(identifier) = Identifier::from_str(&identifier) {
+            identifier
+        } else {
+            return Ok(None);
```

Comment:
maybe we should fail here, someone wrote garbage instead of a correct identifier